pkg/http/api.go

```diff
@@ -975,3 +1001,75 @@ func getMetadataFromRequest(reqCtx *fasthttp.RequestCtx) map[string]string {
 
 	return metadata
 }
+
+func (a *api) onPostStateTransaction(reqCtx *fasthttp.RequestCtx) {
+	var err error
+	if a.stateStores == nil || len(a.stateStores) == 0 {
+		msg := NewErrorResponse("ERR_STATE_STORES_NOT_CONFIGURED", "")
+		respondWithError(reqCtx, 400, msg)
+		return
+	}
+
+	storeName := reqCtx.UserValue(storeNameParam).(string)
+
+	if a.stateStores[storeName] == nil {
+		msg := NewErrorResponse("ERR_STATE_STORE_NOT_FOUND:", fmt.Sprintf("state store name: %s", storeName))
+		respondWithError(reqCtx, 401, msg)
+		return
+	}
+
+	body := reqCtx.PostBody()
+	var requests = []SetDeleteTransactionalRequest{}
+	err = a.json.Unmarshal(body, &requests)
+
+	var transactionalRequests = []state.TransactionalRequest{}
+	for _, request := range requests {
+		var convertedRequest interface{}
+		switch request.Operation {
+		case "delete":
```

Comment:
JFYI - The current gRPC and HTTP API implementation are inconsistent e.g. etag and metadata handling. We will revisit this and refactor in the separate issue and PR.

Reply:
Could you please open a P1 issue for next milestone

Reply:
> Could you please open a P1 issue for next milestone

yes I will put P1 label.